src/lib_version.c

```diff
@@ -0,0 +1,181 @@
+#include "libblastrampoline_internal.h"
+
+#include <ctype.h>
+
+/* We need to ask MKL for information about itself to get better information for the config.
+ *
+ * These are from mkl_types.h
+ */
+typedef struct {
+    int    MajorVersion;
+    int    MinorVersion;
+    int    UpdateVersion;
+    int    PatchVersion;
+    char * ProductStatus;
+    char * Build;
+    char * Processor;
+    char * Platform;
+} MKLVersion;
+
+
+// These are the addresses of the LBT functions for ilaver we export
+extern void ** ilaver_;
+extern void ** ilaver_64_;
+
+// Every library implements their version string handling differently, so this is a ratsnest
+// of conditions for the various libraries to try and get information that is useful to us...
+char* lbt_get_library_info(lbt_library_info_t* library)
+{
+    // Keep the string as a static lifetime so that it never gets deleted
+    int len_info = 512;
+    static char info[512];
```

Comment:
Just to avoid typing 512 twice, should `len_info` be a macro or an enum (`enum { len_info=512 };`) so that you can use it as length of the static array?  Similarly for the other `len_*`  variables below

Reply:
I know a define is supported as an array length specifier with no issues, so I went with that. I guess an enum is probably supported without it turning the array into a VLA, but I've never tried that before.